iocage.8 updates March 2019

Mrt134 opened the pull request:
Update iocage.8 according to changes from these PRs:

- #847 
- #854 
- #867 
- #868 
- #869 
- #878 
- #882 
- #884 

References these FreeNAS Redmine tickets:

- 73740
- 75360
- 75450
- 75468
- 77421
- 77574

Rearchive updated iocage.8
Syntax and flag order will need review.

Make sure to follow and check these boxes before submitting a PR! Thank you.

- [x] Explain the feature
- [x] Read [CONTRIBUTING.md](https://github.com/iocage/iocage/blob/master/CONTRIBUTING.md)

Mrt134 committed:
73740: Add ip_hostname property to iocage.8
local build test: no issues.

Mrt134 committed:
PR 854: Add debug flag to iocage.8

Mrt134 committed:
75360: Add clone_basejail flag to iocage.8 and update basejail flag description.

Mrt134 committed:
75450: Add exec_created property to iocage.8

Mrt134 committed:
75468: Add -f | --force flag to console and exec commands.
PR 869

Mrt134 committed:
77421: Add -f | --force flag to iocage get -P | --plugin option
PR 878

Mrt134 committed:
77574: Add assign_localhost to properties list
PR 882

Mrt134 committed:
77574: Add property localhost_ip to iocage.8
PR 884

Mrt134 committed:
Rearchive iocage.8

Mrt134 committed:
Set manpage date

skarekrow commented on iocage.8:
```diff
@@ -1015,6 +1046,30 @@ The Source listed with each property shows whether it is a local
+Boolean option to add interface
+.Ql lo0
+assigned to
+.Ql 127.0.0.1 .
```
It starts at 127.0.0.2 and works up tot he next available address. 127.0.0.1 is not assigned.

skarekrow commented on iocage.8:
```diff
@@ -1015,6 +1046,30 @@ The Source listed with each property shows whether it is a local
+assigned to
+.Ql 127.0.0.1 .
+Configured automatically when
+.Ql vnet=on.
```
We should reword this as vnet jails setup their own localhost as they are an independent stack. The current text implies vnet=on sets this property. Which isn’t the case

skarekrow reviewed (changes requested):
Looks good! Few changes requested, do you think we can tackle the booleans  changing in this pr as well?

1/0, yes/no, true/false, on/off are all equivalent for many of the properties now. Case does not matter

Mrt134 commented:
Sure, I can update the booleans too, is there a specific syntax you want that applies to all properties?

skarekrow commented:
Perhaps we use 1/0 in the docs and make a note that all of those other values can also apply. Just makes things more consistent.

Mrt134 commented:
Ok, I should have the PR updated in just a bit.

Mrt134 committed:
Update from review:
- Reword descriptions for assign_localhost and localhost_ip to be clearer
- Rework boolean options to use 1 | 0 syntax and add note to PROPERTIES to explain the other options.
- Find and update a few sections to use hopefully clearer macros.
- Rearchive updated manpage.

skarekrow commented on iocage.8:
```diff
@@ -1791,7 +1815,7 @@ Can be set to any string if needed.
 .Pp
 Source: local
-.It Pf compression= Op on | off | lzjb | gzip | gzip-N | zle | lz4
+.It Pf compression= Op 1 | 0 Op lzjb | gzip | gzip-N | zle | lz4
```
Compression and dedup don’t apply with the Boolean changes

Mrt134 replied:
whoops, thanks. I'll get that changed back.

skarekrow commented:
```
            'bpf',
            'template',
            'host_time',
            'basejail',
            'dhcp',
            'vnet',
            'rtsold',
            'jail_zfs',
            'hostid_strict_check',
            'boot',
            'exec_clean',
            'mount_linprocfs',
            'mount_procfs',
            'allow_vmm',
            'allow_tun',
            'allow_socket_af',
            'allow_quotas',
            'allow_mount_zfs',
            'allow_mount_tmpfs',
            'allow_mount_procfs',
            'allow_mount_nullfs',
            'allow_mount_fusefs',
            'allow_mount_devfs',
            'allow_mount',
            'allow_mlock',
            'allow_chflags',
            'allow_raw_sockets',
            'allow_sysvipc',
            'allow_set_hostname',
            'mount_fdescfs',
            'mount_devfs',
            'ip6_saddrsel',
            'ip4_saddrsel',
            'ip_hostname',
            'assign_localhost'
```

Are the boolean props :)

Mrt134 committed:
Fix up compression and dedup properties
Rearchive updated manpage

Mrt134 committed:
Update another couple boolean options and add missing host_time option
Update archive

skarekrow commented:
Thanks Tim!
